engines/shared/app/errors/shared/not_found_error.rb

```diff
@@ -0,0 +1,35 @@
+module Shared
+  class NotFoundError
```

Comment:
As you see, an error is just a duck type that implements the `#status_code`, `#name` and `#message` methods